### DISCUSSION TITLE
is `useWaitForTransaction` replace automatically hash when transaction sped up?

### DISCUSSION THREAD

**Discussion by jeromevvb** [wevm/wagmi#1746] [Q&A] [answered]
Hi everyone,

I was wondering if `useWaitForTransaction` automatically replaced the hash is currently watching with the new hash ?https://wagmi.sh/react/hooks/useWaitForTransaction#onspeedup-optional

**Comment by tmm** [comment, marked as answer]
Yup! See [waitForTransaction.ts#L84](https://github.com/wagmi-dev/wagmi/blob/main/packages/core/src/actions/transactions/waitForTransaction.ts#L84)

**Comment by nfts2me** [comment]
What I see is not actual replacement of the hash, but rather a call to the onSpeedUp / onReplaced callback, right?
